internal/zinc-classpath/src/main/scala/sbt/internal/inc/ScalaInstance.scala

```diff
@@ -123,30 +132,45 @@ object ScalaInstance {
       }
     }
     val jars = provider.jars
-    val loader = provider.loader
     val libraryJar = findOrCrash(jars, "scala-library.jar")
     val compilerJar = findOrCrash(jars, "scala-compiler.jar")
-    new ScalaInstance(version, loader, libraryJar, compilerJar, jars, None)
+    // sbt launcher 1.0.3 will construct layered classloader. Use them if we find them.
+    // otherwise, construct layered loaders manually.
+    val (loader, loaderLibraryOnly) =
```

Comment:
Nitpicks: can we wrap all this logic in curly braces and separate the computation of the default value (what's in `getOrElse`) from the rest of the logic?